geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradePartitionRegionClearServerVersionMismatch.java

```diff
@@ -0,0 +1,171 @@
+            .withProperty(DistributionConfig.ENABLE_CLUSTER_CONFIGURATION_NAME, "false"));
+    final int locatorPort = locator.getPort();
+
+    serverNew = cluster.startServerVM(1, locatorPort);
+    MemberVM serverOld =
+        cluster.startServerVM(2, oldVersion, s -> s.withConnectionToLocator(locatorPort));
+
+    MemberVM.invokeInEveryMember(() -> {
+      Cache cache = getCache();
+      assertThat(cache).isNotNull();
+      getCache().createRegionFactory(RegionShortcut.PARTITION).create("regionA");
+    }, serverNew, serverOld);
+
+    // Put in some boiler plate data for region clear
+    serverNew.invoke(() -> {
+      Cache cache = getCache();
+      assertThat(cache).isNotNull();
+
+      Region<Object, Object> region = cache.getRegion("regionA");
+      region.put("A", "ValueA");
+      region.put("B", "ValueB");
+    });
+
+  }
+
+  /**
+   * testClient_ServerVersionMismatchException - validates that when a client invokes a partitioned
+   * region clear on a cluster where one server is running an unsupported version for this feature
+   * we return a ServerVersionMismatchException
+   */
+  @Test
+  public void testClient_ServerVersionMismatchException() throws Exception {
+    IgnoredException.addIgnoredException(ServerOperationException.class);
+    final int locatorPort = locator.getPort();
+    // Get a client VM
+    ClientVM clientVM = cluster.startClientVM(3, c -> c.withLocatorConnection(locatorPort));
+
+    clientVM.invoke(() -> {
+      // Validate we have a cache and region
+      ClientCache clientCache = getClientCache();
+      assertThat(clientCache).isNotNull();
+
+      ClientRegionFactory<Object, Object> clientRegionFactory =
+          clientCache.createClientRegionFactory(ClientRegionShortcut.PROXY);
+      Region<Object, Object> region = clientRegionFactory.create("regionA");
+      assertThat(region).isNotNull();
+
+      // Validate that we get a ServerVersionMismatchException wrapped in a ServerOperationException
+      Throwable thrown = catchThrowable(region::clear);
+      assertThat(thrown).isInstanceOf(ServerOperationException.class);
+      assertThat(thrown).hasCauseInstanceOf(ServerVersionMismatchException.class);
+
+      // Validate that the message is exactly as we expect it.
+      ServerVersionMismatchException serverVersionMismatchException =
+          (ServerVersionMismatchException) thrown.getCause();
+      assertThat(serverVersionMismatchException.getMessage()).isEqualTo(expectedMessage);
+    });
+  }
+
+  /**
+   * testServer_ServerVersionMismatchException - validates that when a partitioned region clear is
+   * invoked on a cluster where one server is running an unsupported version for this feature we
+   * return a ServerVersionMismatchException
+   */
+  @Test
+  public void testServer_ServerVersionMismatchException() {
+    IgnoredException.addIgnoredException(ServerOperationException.class);
```

Comment:
every test will shut down all the locator/servers and restart them again, making the tests taking longer. If you would like make the tests run faster, you can lump these two tests into one. first all region.clear vm and then in a server vm. And would be nice to verify that after these two clears, the region still has those 2 records in them (clear didn't go through on any servers).

Reply:
Sure, I agree with the record check, but I prefer to have my tests test one thing at time. I realize there is speed efficiency to be had, but I understand that depends on the type of efficiency you are looking for. I am looking at what the test tells me. It certainly feels weird to me though.